plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/PinotQueryBuilder.java

```diff
@@ -131,16 +130,11 @@ private static Optional<String> toPredicate(PinotColumnHandle pinotColumnHandle,
         String predicateArgument = pinotColumnHandle.isAggregate() ? pinotColumnHandle.getExpression() : quoteIdentifier(pinotColumnHandle.getColumnName());
         ValueSet valueSet = domain.getValues();
         if (valueSet.isNone()) {
-            verify(!domain.isNullAllowed(), "IS NULL is not supported due to different null handling semantics. See https://docs.pinot.apache.org/developers/advanced/null-value-support");
+            if (domain.isNullAllowed()) {
+                return Optional.of(format("%s IS NULL", predicateArgument));
+            }
             return Optional.of(format("(%s != %s)", predicateArgument, predicateArgument));
         }
-        if (valueSet.isAll()) {
```

Comment:
iirc this is related to https://github.com/apache/pinot/issues/10600 and 601, since the connector supports 11 and 12.1 should this be put back until we upgrade to 1.0.0 (and remove support for all previous versions)?